synapse/storage/databases/main/events_worker.py

```diff
@@ -780,9 +780,30 @@ async def _get_events_from_cache(
         )
 
         missing_event_ids = {e for e in events if e not in event_map}
```

Comment:
I don't think this needs to be a set?

```suggestion
        missing_event_ids = (e for e in events if e not in event_map)
```